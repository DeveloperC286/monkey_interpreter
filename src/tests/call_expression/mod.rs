use super::*;

#[template]
#[rstest(
    code,
    snapshot_name,
    case("fn(x) { x; }(5);", "call_expression_case_1"),
    case("fn(x) { return x; }(5);", "call_expression_case_2"),
    case("fn(x) { x * 2 }(5);", "call_expression_case_3"),
    case("fn(x, y) { x + y }(2, 3);", "call_expression_case_4"),
    case(
        "let add = fn(x, y) { x + y }; add(1, add(2, 3));",
        "call_expression_case_5"
    )
)]
fn call_expression_cases(code: &str, snapshot_name: &str) {}

#[apply(call_expression_cases)]
fn test_call_expression_lexical_analysis(code: &str, snapshot_name: &str) {
    assert_lexical_analysis!(code, snapshot_name);
}

#[apply(call_expression_cases)]
fn test_call_expression_syntax_analysis(code: &str, snapshot_name: &str) {
    assert_syntax_analysis!(code, snapshot_name);
}

#[apply(call_expression_cases)]
fn test_call_expression_evaluation(code: &str, snapshot_name: &str) {
    assert_evaluation!(code, snapshot_name);
}

#[apply(call_expression_cases)]
fn test_call_expression_environment(code: &str, snapshot_name: &str) {
    assert_environment!(code, snapshot_name);
}
