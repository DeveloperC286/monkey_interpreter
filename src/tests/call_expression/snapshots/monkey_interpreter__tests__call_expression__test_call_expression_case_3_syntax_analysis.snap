---
source: src/tests/call_expression/mod.rs
expression: abstract_syntax_tree
---
[
    Expression {
        expression: Call {
            function: Function {
                parameters: [
                    "x",
                ],
                block: Block {
                    nodes: [
                        Expression {
                            expression: Infix {
                                left_hand: Identifier {
                                    identifier: "x",
                                },
                                operator: Multiply,
                                right_hand: Integer {
                                    literal: 2,
                                },
                            },
                        },
                    ],
                },
            },
            arguments: [
                Integer {
                    literal: 5,
                },
            ],
        },
    },
]
